platform/view/services/db/driver/sql/unversioned.go

```diff
@@ -0,0 +1,146 @@
+/*
+Copyright IBM Corp. All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package sql
+
+import (
+	"database/sql"
+	"fmt"
+
+	"github.com/hyperledger-labs/fabric-smart-client/platform/view/services/db/driver"
+)
+
+type Unversioned struct {
+	base
+}
+
+func NewUnversioned(db *sql.DB, table string) *Unversioned {
+	return &Unversioned{
+		base: base{
+			db:    db,
+			table: table,
+		},
+	}
+}
+
+func (db *Unversioned) SetState(ns, key string, val []byte) error {
+	if len(val) == 0 {
+		logger.Warnf("set key [%s:%s] to nil value, will be deleted instead", ns, key)
+		return db.DeleteState(ns, key)
+	}
+	if db.txn == nil {
+		panic("programming error, writing without ongoing update")
+	}
+	logger.Debugf("set state [%s,%s]", ns, key)
+
+	val = append([]byte(nil), val...)
+
+	// Portable upsert
+	exists, err := db.exists(db.txn, ns, key)
+	if err != nil {
+		return err
+	}
+	if exists {
+		query := fmt.Sprintf("UPDATE %s SET val = $1 WHERE ns = $2 AND pkey = $3", db.table)
+		logger.Debug(query, len(val), ns, key)
+
+		_, err := db.txn.Exec(query, val, ns, key)
+		if err != nil {
+			return fmt.Errorf("could not set val for key [%s]: %w", key, err)
+		}
+	} else {
+		query := fmt.Sprintf("INSERT INTO %s (ns, pkey, val) VALUES ($1, $2, $3)", db.table)
+		logger.Debug(query, ns, key, len(val))
+
+		_, err := db.txn.Exec(query, ns, key, val)
+		if err != nil {
+			return fmt.Errorf("could not insert [%s]: %w", key, err)
+		}
+	}
+
+	return nil
+}
+
+func (db *Unversioned) GetState(ns, key string) ([]byte, error) {
+	var val []byte
+
+	query := fmt.Sprintf("SELECT val FROM %s WHERE ns = $1 AND pkey = $2", db.table)
+	logger.Debug(query, ns, key)
+
+	row := db.db.QueryRow(query, ns, key)
+	if err := row.Scan(&val); err != nil {
+		if err == sql.ErrNoRows {
+			logger.Debugf("not found: [%s:%s]", ns, key)
+			return val, nil
+		}
+		return val, fmt.Errorf("error querying db: %w", err)
+	}
+
+	return val, nil
+}
+
+func (db *Unversioned) GetStateRangeScanIterator(ns string, startKey string, endKey string) (driver.ResultsIterator, error) {
+	where := ""
+	args := []interface{}{ns}
+
+	// To match badger behavior, we don't include the endKey
+	if startKey != "" && endKey != "" {
```

Comment:
You could extract this string creation into a util function in case you need it again.